src/Test.UnitTests.Sarif.Driver/FunctionlessTestRule.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.IO;
+using System.Resources;
+using Microsoft.CodeAnalysis.Sarif.Driver;
+using Microsoft.CodeAnalysis.Test.UnitTests.Sarif.Driver.Sdk;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    /// <summary>
+    /// This test class exists to provide an additional check that can be disabled via context, etc. It has no configurable behavior.
+    /// </summary>
+    [Export(typeof(ReportingDescriptor)), Export(typeof(IOptionsProvider)), Export(typeof(Skimmer<TestAnalysisContext>))]
+    internal class FunctionlessTestRule : TestRuleBase, IOptionsProvider
+    {
+        private const string FunctionlessTestRuleId = "TEST1002";
+
+        public override string Id => FunctionlessTestRuleId;
+
+        protected override ResourceManager ResourceManager => SkimmerBaseTestResources.ResourceManager;
+
+        protected override IEnumerable<string> MessageResourceNames => new List<string>
+        {
+        };
+
+        public override MultiformatMessageString FullDescription { get { return new MultiformatMessageString { Text = "This is the full description for TEST1002" }; } }
+
+        public override void Analyze(TestAnalysisContext context)
+        {
+        }
+        
+        public IEnumerable<IOption> GetOptions()
+        {
+            return new IOption[] { Behaviors, UnusedOption };
+        }
+
+        private const string AnalyzerName = FunctionlessTestRuleId + "." + nameof(FunctionlessTestRule);
+
+        public static PerLanguageOption<TestRuleBehaviors> Behaviors { get; } =
+            new PerLanguageOption<TestRuleBehaviors>(
+                AnalyzerName, nameof(TestRuleBehaviors), defaultValue: () => { return TestRuleBehaviors.None; });
```

Comment:
>{ [](start = 77, length = 1)

`() => TestRuleBehaviors.None` should work; saves a few characters.

Reply:
applied the same note to other opportunities in this file as well, thanks.

---
In reply to: [322891605](https://github.com/microsoft/sarif-sdk/pull/1670#discussion_r322891605) [](ancestors = 322891605)